src/Atlas2DSlice.ts

```diff
@@ -0,0 +1,53 @@
+import { BufferGeometry, Material, Mesh, PlaneGeometry, ShaderMaterial, ShapeGeometry } from "three";
+import { Volume } from ".";
+import { sliceFragmentShaderSrc, sliceVertexShaderSrc } from "./constants/volumeSliceShader";
+import RayMarchedAtlasVolume from "./RayMarchedAtlasVolume";
+import { rayMarchingShaderUniforms } from "./constants/volumeRayMarchShader";
+
+/**
+ * Creates a plane that renders a 2D XY slice of volume atlas data.
+ */
+export default class Atlas2DSlice extends RayMarchedAtlasVolume {
+  constructor(volume: Volume) {
+    super(volume);
+    this.setUniform("Z_SLICE", Math.floor(volume.z / 2));
+  }
+
+  // TODO: Change uniforms so that it can be a different/custom type
+  // (this requires a refactor of some of the render implementation classes)
+  // Overload geometry to create plane instead of a cube mesh
+  protected createGeometry(
+    uniforms: typeof rayMarchingShaderUniforms
+  ): [ShapeGeometry, Mesh<BufferGeometry, Material>] {
+    const geom = new PlaneGeometry(1.0, 1.0);
+    const mesh: Mesh<BufferGeometry, Material> = new Mesh(geom);
+    mesh.name = "Plane";
+
+    // shader,vtx and frag.
+    const vtxsrc = sliceVertexShaderSrc;
+    const fgmtsrc = sliceFragmentShaderSrc;
+
+    const threeMaterial = new ShaderMaterial({
+      uniforms: uniforms,
+      vertexShader: vtxsrc,
+      fragmentShader: fgmtsrc,
+      transparent: true,
+      depthTest: true,
+      depthWrite: false,
+    });
+
+    mesh.material = threeMaterial;
+    return [geom, mesh];
+  }
+
+  public setZSlice(_slice: number): boolean {
```

Comment:
code style:  this is not strictly enforced, but usually if a function arg is prefixed with underscore it is indicating that it's unused.   Typically to appease typescript/eslint.

Reply:
Yup, `_slice` is marked with an underscore here because it's unused in some implementations of `VolumeRenderImpl`.

Reply:
In this case I would name the argument with an underscore only in implementations where the field is unused. Argument names don't have to match between interfaces and implementations as long as their types do.